File: Segundo/Monopoly/civitas_ruby (1)/proyecto_civitas_ruby/lib/casilla.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
#require_relative 'diario'
#require_relative 'tipo_casilla'
#require_relative 'titulo_propiedad'
#require_relative 'mazo_sorpresas'
#require_relative 'sorpresa'
module Civitas
 class Casilla
   
    def initialize(nombre)
      @nombre= nombre
    end
    public
    def getNombre
      return @nombre
    end
    
    def getTituloPropiedad()
      return @tituloPropiedad 
    end
    public
    def jugadorCorrecto(actual,todos)
      return actual<=todos.size
    end
    private
    def informe( actual,todos)
      if jugadorCorrecto(actual,todos) then
        msg= "Nombre: "+todos.at(actual).getNombre()+"en la casilla"+ toString()
        Diario.instance.ocurre_evento(msg)
      end
    end
    
  public
  def toString
    msg="\n\nCasilla:\n\tNombre:  #{@nombre}\n\n"
    return msg
  end
  
  def recibeJugador (actual,todos)
    informe(actual,todos)
#    if (@tipo == TipoCasilla::SORPRESA)
#      recibeJugador_sorpresa(actual,todos)
#      
#    else
#      if (@tipo == TipoCasilla::JUEZ)
#      recibeJugador_juez(actual , todos )
#      
#      else
#        
#        if (@tipo == TipoCasilla::IMPUESTO)
#        recibeJugador_impuesto(actual , todos ) 
#        else
#          if (@tipo == TipoCasilla::CALLE)
#          recibeJugador_calle( actual,todos)
#          else
#            informe(actual,todos)
#          end          
#          
#        end      
#      end    
    
      
  end
    
    
    
    
    
    
    

  
#  def recibeJugador_calle( actual,todos)
#    if jugadorCorrecto(actual,todos)
#      informe(actual,todos)
#      jugador=todos.at(actual)
#      if (@tituloPropiedad.tienePropietario()==false)
#        jugador.puedeComprarCasilla
#      else
#        @tituloPropiedad.tramitarAlquiler(jugador)
#        
#        
#      end
#    end
#  end
#  def recibeJugador_sorpresa( actual,todos)
#    if jugadorCorrecto(actual,todos)
#      sorpresa = @mazo.Siguiente 
#      informe(actual,todos)
#      sorpresa.aplicarAJugador(actual,todos)
#      
#    end
#  end
#  def recibeJugador_impuesto(actual , todos ) 
#    if jugadorCorrecto(actual,todos)
#      informe(actual,todos)
#      todos.at(actual).pagaImpuesto(@importe )
#    end
#  end
#  def recibeJugador_juez(actual , todos )
#    if jugadorCorrecto(actual,todos)
#      informe(actual,todos)
#      todos.at(actual).encarcelar(@carcel) 
#    end
#    
#  end 
  

  
 private :informe   
 end
end

File: Segundo/Monopoly/civitas_java (1)/proyecto_civitas/src/civitas/Casilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

/**
 *
 * @author pc
 */
import java.util.ArrayList;
public class Casilla{
    private String nombre;

 
    public Casilla(String nombre){
        
        this.nombre= nombre;
    
    }
    public String getNombre() {
        
        return nombre;
    }
    public void setCasilla(String nombre){
        
        this.nombre= nombre;
    
    }
    public void informe (int actual, ArrayList<Jugador> todos){
        String msg;
        msg=("Nombre: "+todos.get(actual).getNombre()+" en la Casilla "+toString());
        Diario.getInstance().ocurreEvento(msg);
    }
    @Override
    public String toString() {
        return "Casilla Descanso" + "\n      Nombre : " + nombre + '\n';
    }
    
    public boolean jugadorCorrecto(int actual, ArrayList<Jugador> todos){
        return actual<=todos.size();
    }
    public void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos))
        informe(actual,todos);
    }
}